sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobContainerAsyncClient.java

```diff
@@ -1045,4 +1045,47 @@ private boolean validateNoETag(BlobRequestConditions modifiedRequestConditions)
         }
         return modifiedRequestConditions.getIfMatch() == null && modifiedRequestConditions.getIfNoneMatch() == null;
     }
+
+    /**
+     * Restores a previously deleted container.  The restored container
+     * will be renamed to the name of this {@link BlobContainerAsyncClient}.
+     * If the container associated with this {@link BlobContainerAsyncClient}
+     * already exists, this call will result in a 409 (conflict).
+     * This API is only functional if Container Soft Delete is enabled
+     * for the storage account associated with the container.
+     *
```

Comment:
The docs should point to the REST docs and should have a code snippet.

Reply:
I added snippets and left TODO for links as we don't have them yet.